lms/djangoapps/courseware/views/views.py

```diff
@@ -513,6 +513,13 @@ def create_page_context(self, request, course=None, tab=None, **kwargs):
             request.user = masquerade_user
         else:
             masquerade = None
+
+        now = datetime.now(UTC())
+        effective_start = _adjust_start_date_for_beta_testers(request.user, course, course.id)
+        if not in_preview_mode() and staff_access and now < effective_start:
```

Comment:
Additionally, once this logic has been hammered out, it seems like this should have unit tests.